velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -66,4 +69,46 @@ const TimestampToStringOptions& SparkCastHooks::timestampToStringOptions()
 bool SparkCastHooks::truncate() const {
   return true;
 }
+
+std::string SparkCastHooks::castMapToString(
+    const MapVector* mapVector,
+    const int row) const {
+  auto keys = mapVector->mapKeys();
+  auto values = mapVector->mapValues();
+  auto rawOffsets = mapVector->rawOffsets();
+  auto rawSizes = mapVector->rawSizes();
+
+  // spark.sql.legacy.castComplexTypesToString.enabled
+  // When true, maps and structs are wrapped by [] in casting to strings, and
+  // NULL elements of structs/maps/arrays will be omitted while converting to
+  // strings. Otherwise, if this is false, which is the default, maps and
+  // structs are wrapped by {}, and NULL elements will be converted to "null".
```

Comment:
Could we support configurable behaviors in the implementation of `mapVector->toString`, and return specific configs from cast hooks?

Reply:
Good idea. Let me probe it.

Reply:
Thanks for your advice. I added a toString with options function so that the conversion to string can be configured based on options. Further array/struct conversions can be implemented based on this function in the future.